sys/cbor/cbor.c

```diff
@@ -400,7 +400,7 @@ static size_t decode_bytes(const cbor_stream_t *s, size_t offset, char *out, siz
         return 0;
     }
 
-    if (length + 1 < bytes_length) {
+    if (bytes_length >= length) {
```

Comment:
I think your are right that this check is wrong, but a _better_ fix would be to rewrite to `if (length < bytes_length + 1) {` - that way keeping the notion of the original code and `<` is simpler than `>=` though the compiler will likely optimize that the same way.

I do not know CBOR in detail, but I was a little surprised that the output buffer it terminated with `\0`, as it also knows the length of the string from the return value.

Reply:
Ok I changed it. 
Most of the cbor functions return the number of bytes read from the cbor stream.